Bucketeer/Sources/Public/BKTEvaluation.swift

```diff
@@ -19,3 +21,64 @@ public struct BKTEvaluation: Equatable {
         case prerequisite = "PREREQUISITE"
     }
 }
+
+public struct BKTEvaluationDetails<T:Equatable>: Equatable {
+    public let featureId: String
+    public let featureVersion: Int
+    public let userId: String
+    public let variationId: String
+    public let variationName: String
+    public let variationValue: T
+    public let reason: Reason
+
+    public enum Reason: String, Codable, Hashable {
```

Comment:
Although it is similar to the [ReasonType](https://github.com/bucketeer-io/ios-client-sdk/blob/main/Bucketeer/Sources/Internal/Model/ReasonType.swift#L3C6-L3C16) definition, it is not very desirable to have the same definition in multiple places, so I would like you to consider using it uniformly if possible. Or consider removing the definition if you don't need it.

Reply:
[ReasonType](https://github.com/bucketeer-io/ios-client-sdk/blob/main/Bucketeer/Sources/Internal/Model/ReasonType.swift#L3C6-L3C16)

is for internal use only. I maintain the same approach like before

Reply:
I would like to keep this. The Android SDK has something like this as well. @kakcy